### PR TITLE
Make dunder lookup use tp$getattr() again. Gives us better error messages.

### PR DESCRIPTION
**Pull request by meredydd**
tl;dr Fix "this object is not callable"/"not iterable" error messages.

This mostly reverses commit ae9fed464, which tried to accommodate dynamically provided dunder-functions by using Sk.abstr.gattr(). The correct fix (implemented in commit 2e43d47 / PR #634) was to make tp$getattr() do dynamic lookup properly. Using Sk.abstr.gattr() just made things complicated, and in particular made `__call__`/`__iter__` lookup failures come out as generic AttributeErrors rather than TypeErrors with helpful messages.

Reversing this means that we get better error messages (like "this object is not callable/iterable") again.

### CODE REVIEW

**Comment by waywaaard** [comment]
How should it be implemented? Is this the right behavior or just a fix for the newly introduced errors?

**Comment by meredydd** [comment]
The behaviour was, and is, correct - this patch brings no functional changes. The only change is that the code is a bit simpler and the errors are more human-friendly.

**Comment by meredydd** [comment]
@waywaard poke?

**Comment by albertjan** [comment]
Looks good to me.

**Comment by waywaaard** [comment]
I am not allowed to vote^^. Sorry for the late reply.

**Comment by bnmnetp** [comment]
@waywaaard I am happy to grant you voting/committing rights.  Last time I asked I did not get a response...

**Comment by waywaaard** [comment]
@bnmnetp I am sorry to hear, that you did not receive a response. Though, I am not sure where you send the message.
